kube-runtime/src/controller/mod.rs

```diff
@@ -319,6 +320,11 @@ where
                     None => future::err(Error::ObjectNotFound(request.obj_ref.erase())).right_future(),
                 }
             })
+            .delay_tasks_until(async move {
```

Comment:
Ok, I realized the problem..

`reconciler` is a `FnMut`, so to be able to run its future concurrently we need to call `reconciler()` _before_ we schedule anything to run in parallel. Moving it into the async context would violate that constraint. I'm not sure we have a good reason for it to be `FnMut` (rather than `Fn`), but we can't change that without a breaking change.